### PR TITLE
Create Scientific Metadata Search Engine for PostgreSQL (FullText)

### LINKED ISSUE #457: Support FullText search in PostgreSQL

### PR DESCRIPTION
**Pull request by Kezzsim**
fix #457

This does a number of things to provide basic $FullText search that is idempotent to mongoDB without using any third party indexers. Utilizes Postgres's built in functions [ts_vector & ts_query](https://www.postgresql.org/docs/current/functions-textsearch.html). Both of these methods appear to support `jsonb` columns out of the box.

Progress:
- [x] Create `metadata_search` index during `tiled catalog init` in `orm.py`
- [x] Add fullText as a query method to `adapter.py`
- [ ] Add fullText logic to `adapter.py`
- [ ] Create Alembic migration to add `metadata_search` index to existing Tiled deployments
- [ ] Enable fullText pytests

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206267196412362

### CODE REVIEW

File: tiled/catalog/adapter.py [hunk 1]
```diff
@@ -975,6 +976,11 @@ def contains(query, tree):
     return tree.new_variation(conditions=tree.conditions + [condition])
 
 
+def fullText(query, tree):
```
Comment (danielballan) [inline tiled/catalog/adapter.py:979]: [PEP 8](https://peps.python.org/pep-0008/#function-and-variable-names):

> Function names should be lowercase, with words separated by underscores as necessary to improve readability.

> Variable names follow the same convention as function names.

> mixedCase is allowed only in contexts where that’s already the prevailing style (e.g. threading.py), to retain backwards compatibility.

**Comment by danielballan** [comment]
While beginning to try to understand this feature for review, I came across an article that you may have also found: https://amitosh.medium.com/full-text-search-fts-with-postgresql-and-sqlalchemy-edc436330a0c

The approach declares a "computed" column and runs searches against that. I wonder how this compares to the PR’s approach, declaring an index with:

```py
func.to_tsvector(text("'simple'"), func.lower("metadata")),
```

**Comment by danielballan** [comment]
Ah, indeed; I read the Issue and the PR out of order. You found the same article. Worth a close look, I think.

File: tiled/catalog/adapter.py (hunk 1 above)
Comment (Kezzsim) [reply tiled/catalog/adapter.py]: Added in [b713504fddbb7f0c064e27900dbcc97473365ceb](https://github.com/bluesky/tiled/pull/628/commits/b713504fddbb7f0c064e27900dbcc97473365ceb)

**Pull request closed without merging**